core/module/services.types.ts

```diff
@@ -54,10 +54,12 @@ export interface AuthService {
 }
 
 export interface SubscriptionService {
-  getSubscriptionLink(): Promise<{ url: string | null }>;
+  getSubscriptionLink(): Promise<{ url: string | undefined }>;
+  getManageSubscriptionLink(): Promise<{ url: string | undefined }>;
   getActiveSubscription(): Promise<SubscriptionResponse | null>;
   getInvoices(
     from: number,
     to: number
   ): Promise<{ invoices: GenesisInvoice[]; count: number }>;
+  cancelSubscription(): Promise<Record<string, string | number>>;
```

Comment:
Lets avoid this, is super hard to refactor later
`Record<string, string | number>` I believe is enough we return a boolean in the interface

Reply:
I am not sure how to go about this but we have a async operation inside the function which must return a Promise.